src/crypto/equihash.cpp

```diff
@@ -230,12 +230,12 @@ FullStepRow<WIDTH>::FullStepRow(const unsigned char* hashIn, size_t hInLen,
 }
 
 template<size_t WIDTH> template<size_t W>
-FullStepRow<WIDTH>::FullStepRow(const FullStepRow<W>& a, const FullStepRow<W>& b, size_t len, size_t lenIndices, int trim) :
+FullStepRow<WIDTH>::FullStepRow(const FullStepRow<W>& a, const FullStepRow<W>& b, size_t len, size_t lenIndices, size_t trim) :
         StepRow<WIDTH> {a}
 {
     assert(len+lenIndices <= W);
     assert(len-trim+(2*lenIndices) <= WIDTH);
```

Comment:
Just a reminder. I didn't check all the upstreams but changing `int` to `size_t` may introduce potential integer overflow when doing subtraction.

As an example: a simple `for (int i = 0; i < n; i++)` to `for (size_t i = 0; i < n; i++)` won't have problem but `for (size i = n-1; i >=0; i--)` is an error.

However in this case, it looks like everything is fine.